### ISSUE TITLE
Invalid state after changing animation

### ISSUE THREAD

**Issue by esavichev** [pixijs/pixi-spine#129]
This example illustrates the bug. https://github.com/pixijs/pixi-spine/files/577754/pixi_spine_issue127.zip

Run index.html
Hover first animation "idle". This will trigger `setAnimation(0, "fright", true)`. This animation uses big white eyes which are invisible in other animations.
Hover first animation again. This will trigger `setAnimation(0, "idle", true)`. In this animation big white eyes should be invisible, but they are still visible after previous animation.
Any new calls to `setAnimation` won't update those elements visibility.

And one of the eyes is positioned wrong (looks like it is related to #128)

Latest spine-webgl runtime from spine-ts dev branch displays this animation correctly (see spine-ts-demo.html, click to switch animations).

**Comment by ivanpopelyshev** [comment]
Gonna look at it today

**Comment by ivanpopelyshev** [comment]
Gonna look at it today

**Comment by esavichev** [comment]
Thanks, feel free to ask for anything else necessary to test or reproduce the bugs.

**Comment by esavichev** [comment]
After quick research looks like the issue is located. The problem is in Spine.update method. There are 2 possible types of attachment, and if the type changes through different animations, the other type is not being cleaned up.

The next experiment in Spine.ts fixes the issue: 

Line 205
```
if (slot.currentMesh) {
    slot.currentMesh.visible = false;
    slot.currentMesh = null;
    slot.currentMeshName = undefined;
}
```

Line 259
```
if (slot.currentSprite) {
    slot.currentSprite.visible = false;
    slot.currentSprite = null;
    slot.currentSpriteName = undefined;

    if (slotContainer.transform) {
        slotContainer.transform = new PIXI.TransformBase();
    }
    else {
        slotContainer.localTransform = new PIXI.Matrix();
        slotContainer.displayObjectUpdateTransform = PIXI.DisplayObject.prototype.updateTransform;
    }
}
```

Not sure if this is complete solution, maybe there's something else I did not take in account or it is a kind of hack and can be implemented more accurate. Anyway, the issue starts here.

PS: this fixes #128 as well.

**Comment by ivanpopelyshev** [comment]
Please check that new version works.

**Comment by esavichev** [comment]
Yes, the new version works as expected, thanks. And this fix also closes #128 which has the same roots.

The only thing you should keep in mind, that possibly this solution is not full or correct for any case, as it was just a fast experiment without deep digging into spine runtime and data structures.

**Comment by ivanpopelyshev** [comment]
@esavichev no, i believe that your solution is correct. Spine.ts is one big hack ;) It will be easier in the next version of pixi, i want ability to "lock" transforms without re-creating them
